### DISCUSSION TITLE
Charge for two hours

### DISCUSSION THREAD

**Discussion by NicLogix** [celsworth/lxp-bridge#220] [Ideas]
I am using Home Assistant add-on for lxp-bridge.
Being a South African resident, we have a crazy schedule of load-shedding, having no power 2 - 3 times a day, for 2 or 4 hours.

My goal is to have [Quick charge 2 hrs] button or script to do a 2hr ac charge/ until battery on acceptable level.

I was wondering If I can request the following:
I am able to read the AC charge time nr3.
I can save the times in json format. (As it is read in lxp-bridge, with "start" & "end")
I can post custom time (from now - now+2hrs) in json format with mqtt.
*** I cannot send this info back to inverter/dongle ***
I monitor battery level and time with automation.
If 2 hrs passed / battery level is right, I can post message with mqtt with the saved ac charge nr3 times.

If it is possible to have mqtt topic to post the ac charge 3 times to add-on, which in turn send this info to inverter, then repost lxp/BAxxxxxxxx/inputs/all to ensure times have been updated.

I have been looking at the TCP spec, but cannot wrap my head around it yet.

As always, fantastic work with lxp-bridge.

**Comment by jacovt** [comment]
Hello @NicLogix ,

As your fellow South African's feeling your pain, we have already solved that over here: ;)
https://github.com/celsworth/lxp-bridge/discussions/180

Thanks,
---Jaco

**Comment by NicLogix** [comment]
I have to apologise, last time I read that, the guys were just toggling the AC Charge on and off.

I tried to formulate and post a mqtt message to lxp/cmd/BAxxxxxxxx/set/ac_charge/3 as {"start":"00:00"... but with no success.
I honestly thought this had to be sent to the inverter as a hex string, as well as posting directly to the holding registers with a hex string  (facepalm!).

I tested posting to lxp/cmd/BA24702349/set/hold/72 hours and on [73] minutes and this is fully working.  Will just change my scripts to grab hours and minutes and post individually.

FYI, Accuweather integration for Home Assistant predicts the amount of sunshine hours for current day + 4 days ahead, which also influences battery levels with load shedding.

I am also using 
https://github.com/DanteWinters/lux-power-distribution-card
for a good looking "Luxpower" frontend in lovelace.

Thank you for pointing me in the right direction.
@celsworth even better than expected, again thank you.
